s4e-backend/src/main/java/pl/cyfronet/s4e/service/InvitationService.java

```diff
@@ -92,6 +85,13 @@ public void remove(String token) throws NotFoundException {
         return invitationRepository.findByToken(token, projection);
     }
 
+    public void deleteBy(String token) {
+        val invitation = invitationRepository.findByToken(token, Invitation.class);
+        if (invitation.isPresent()) {
+            invitationRepository.delete(invitation.get());
+        }
+    }
```

Comment:
Shouldn't this throw NFE exception in case invitation is not found? Also, the method should be transactional, as you're doing two operations (find and delete) here.